plugins/zenoh-plugin-storage-manager/src/memory_backend/mod.rs

```diff
@@ -246,6 +251,15 @@ impl Storage for MemoryStorage {
         }
         Ok(())
     }
+
+    async fn get_all_entries(&self) -> ZResult<Vec<(String, Timestamp)>> {
+        let mut result = Vec::new();
+        let map = self.map.read().await;
+        for k in map.keys() {
```

Comment:
It would be more efficient to loop on `map.iter()` to get each entry. That would avoid the `map.get()` for each key.